src/containers/PipelineRuns/PipelineRuns.js

```diff
@@ -124,11 +140,53 @@ export function PipelineRuns({ intl }) {
     });
   }
 
+  function persistCancelState(state) {
+    setCancelState(state);
+    setDefaultCancelState(state);
```

Comment:
ideally we would only be setting this in one place and reading from that single place consistently.

We could use a similar to pattern to the one in [ListPageLayout](https://github.com/tektoncd/dashboard/blob/136c2d99483313b756321ea107b5ba024d2c862d/src/containers/ListPageLayout/ListPageLayout.js#L61-L63) for the pagination settings. Set the value in `state`, listen to `state` updates and write to `localStorage`. This way they should always be in sync and there's one trigger for the update.

We would read from `localStorage` on mount and use it to populate the initial value in `state`. From that point on `state` is our source of truth, until the page is reloaded.